sklearn/impute.py

```diff
@@ -1207,6 +1208,17 @@ def fit(self, X, y=None):
             raise ValueError("'features' has to be either 'missing-only' or "
                              "'all'. Got {} instead.".format(self.features))
 
+        if isinstance(self.sparse, bool) and self.sparse and self.missing_values == 0:
+            raise ValueError("'missing_values' can not be 0 when 'sparse' is True")
+
```

Comment:
I don't think we want this case to raise an error.

Reply:
ok, I removed this case. Thanks!